### ISSUE TITLE
bug in COCO metrics callback and tensorflow backend NMS cause recall are much higher than precision and multiple bboxes on the same object

### ISSUE THREAD

**Issue by hulkds** [keras-team/keras-cv#2526]
## Current Behavior:
When running https://keras.io/examples/vision/yolov8/ tutorial you may encountered the problem where the recall are much higher than precision, and there are too many same bounding boxes for the same object, this is because:

PyCOCOCallback computes the number of detections within annotations as follows: 
https://github.com/keras-team/keras-cv/blob/94b0a551d03ee9de3d81663e2c1a680cb113f7f1/keras_cv/src/callbacks/pycoco_callback.py#L102 

The TensorFlow backend is using `tf.image.non_max_suppression_padded` to compute NMS, but it doesn't work as expected—the `iou_threshold` parameter is not taken into account. Whether it's set to 0.1 or 0.9, the behavior remains the same:

https://github.com/keras-team/keras-cv/blob/94b0a551d03ee9de3d81663e2c1a680cb113f7f1/keras_cv/src/layers/object_detection/non_max_suppression.py#L92-L100

## Expected Behavior:
I fixed these bugs in my fork as following:

Use `>=` when counting `num_detection`
[`num_detections = ops.sum(ops.cast(gt_classes >= 0, "int32"), axis=-1)` ](https://github.com/hulkds/keras-cv/blob/3f06de897aeecfcdf43507a86a1b70af6c6bef9d/keras_cv/src/callbacks/pycoco_callback.py#L102)

Use keras_cv implemented `non_max_supression` function instead of `tf.image.non_max_suppression_padded`:
https://github.com/keras-team/keras-cv/blob/94b0a551d03ee9de3d81663e2c1a680cb113f7f1/keras_cv/src/layers/object_detection/non_max_suppression.py#L186

## Version:
"keras-cv==0.9.0",
"keras==3.10.0",
"keras_hub==0.18.1",
"tensorflow[and-cuda]==2.18.0", 

## Anything else:
Must use `bounding_box_format = 'yxyx'` if using keras_cv non_max_suppression function as described in function docstring.

My fork: https://github.com/hulkds/keras-cv/tree/fix_nms_and_coco_metric

I am available for a PR

**Comment by sachinprasadhs** [comment]
@hulkds Thanks for reporting the issue! We have consolidated the development of KerasCV into the new **KerasHub** package, which supports image, text, and multi-modal models). `KerasHub` will support all the core functionality of `KerasCV`.

`KerasHub` can be installed with `!pip install -U keras-hub`. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
